backend/src/test/java/ai/verta/modeldb/BranchTest.java

```diff
@@ -672,16 +679,16 @@ public void branchTest() {
         commitShaList.get(3),
         branchRootCommit.getCommitSha());
 
-    ListBranchCommitsRequest listBranchCommitsRequest =
-        ListBranchCommitsRequest.newBuilder()
+    ListCommitsLogRequest listCommitsLogRequest =
+        ListCommitsLogRequest.newBuilder()
             .setRepositoryId(RepositoryIdentification.newBuilder().setRepoId(id).build())
             .setBranch(branchName)
             .build();
-    ListBranchCommitsRequest.Response listBranchCommitsResponse =
-        versioningServiceBlockingStub.listBranchCommits(listBranchCommitsRequest);
+    ListCommitsLogRequest.Response listBranchCommitsResponse =
+        versioningServiceBlockingStub.listCommitsLog(listCommitsLogRequest);
```

Comment:
`listCommitsLog` returns all the child commits or all the parent commits? because as per my understanding `commit A` --> `commit B` --> `commit C` --> `commit D` where I have marked `commit C` as a branch `XYZ` then when I called listCommitsLog what should expected result? is it `commit C` --> `commit D` ? OR is it `commit A`, `commit B` & `commit C`?

Reply:
commit A, commit B & commit C?

Reply:
Ok then I have to change the logic according to your given resultset

Reply:
This was my confusion. It works as expected.